OpenRA.Mods.Common/Activities/Repair.cs

```diff
@@ -98,7 +98,7 @@ public override Activity Tick(Actor self)
 
 			if (remainingTicks == 0)
 			{
-				var unitCost = self.Info.TraitInfo<ValuedInfo>().Cost;
+				var unitCost = self.Info.HasTraitInfo<ValuedInfo>() ? self.Info.TraitInfo<ValuedInfo>().Cost : 0;
```

Comment:
Use `var valuedInfo = self.Info.TraitInfoOrDefault<ValuedInfo>()` and then do a null check instead of performing two lookups!

This same comment applies to the other changes in the commit too.